File: 평강/app/project/src/main/java/com/eventpurchase/project/domain/product/repository/ProductRepository.java
package com.eventpurchase.project.domain.product.repository;

import com.eventpurchase.project.domain.product.entity.ProductEntity;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface ProductRepository extends JpaRepository<ProductEntity, Long> {

    // lastId보다 작은 id를 내림차순으로 size만큼 가져옴
    List<ProductEntity> findByIdLessThanOrderByIdDesc(Long lastId, Pageable pageable);

}

File: 평강/app/project/product/infrastructure/src/main/kotlin/com/eventpurchase/project/product/ProductRepository.kt
package com.eventpurchase.project.product

interface ProductRepository {
    fun findById(id: Long): Product?
    fun findAll(size: Int): List<Product>
    fun findByIdLessThan(lastId: Long, size: Int): List<Product>
    fun count(): Long
    fun saveAll(products: List<Product>)
}
